### PR TITLE
Test Flake + r/rw handling in form

### LINKED ISSUE #5335: Canvas test flake

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

Fixed test flake, added functionality to block hide r/rw values from form, it works the same as in normal form, also added functionality to hide deleting connection button on connection between r/rw embedded entities and its parent

closes #5335 #5345 #5111 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

File: src/UI/Components/Diagram/helpers.ts [hunk 1]
```diff
@@ -173,18 +184,23 @@ export const checkIfConnectionIsAllowed = (
 /**
  * Iterate through connectedElements of some shape to check if there are possible connections left for given shape
  *
- * @param {ServiceEntityBlock[]} connectedElements
- * @param {Rule | undefined} rule
+ * @param {ServiceEntityBlock[]} connectedElements llist of connected elements to given shape
```
Comment (LukasStordeur) [inline src/UI/Components/Diagram/helpers.ts:187]:
```suggestion
 * @param {ServiceEntityBlock[]} connectedElements list of connected elements to given shape
```

File: src/UI/Components/Diagram/helpers.ts [hunk 2]
```diff
@@ -173,18 +184,23 @@ export const checkIfConnectionIsAllowed = (
...
+ * @param {Rule | undefined} rule rules telling which shapes can connect to eachother and about their limitations
```
Comment (LukasStordeur) [inline src/UI/Components/Diagram/helpers.ts:188]:
```suggestion
 * @param {Rule | undefined} rule telling which shapes can connect to each other and about their limitations
```

File: src/UI/Components/Diagram/helpers.ts [hunk 3]
```diff
@@ -173,18 +184,23 @@ export const checkIfConnectionIsAllowed = (
...
 const checkWhetherConnectionRulesAreExhausted = (
   connectedElements: ServiceEntityBlock[],
   rule: Rule | undefined,
+  editMode: boolean,
 ): boolean => {
   const targetConnectionsForGivenRule = connectedElements.filter(
     (element) => element.getName() === rule?.name,
   );
 
+  //if is in edit mode and it's modifier is r/rw then the connections are basically exhausted
```
Comment (LukasStordeur) [inline src/UI/Components/Diagram/helpers.ts:200]:
```suggestion
  //if is in edit mode and its modifier is r/rw then the connections are basically exhausted
```

File: src/UI/Components/Diagram/interfaces.ts [hunk 4]
```diff
@@ -20,8 +20,10 @@ interface DictDialogData {
 }
 interface Rule {
   name: string;
+  type: "embedded" | "inter-service";
```
Comment (LukasStordeur) [inline src/UI/Components/Diagram/interfaces.ts:23]: Like mentioned previously, I think we can use a Type/Enum for this.

File: src/UI/Components/Diagram/actions.ts [hunk 5]
```diff
@@ -419,6 +463,7 @@ export function appendEntity(
   entity: InstanceAttributeModel,
   isCore: boolean,
   isEmbedded = false,
+  holderType = "",
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/UI/Components/Diagram/actions.ts:466]: As you asked I now changed that variable to holderName, essentially it's name of the entity to which embedded entity belongs to

[the remark this answers is not in the document]
Comment (LukasStordeur) [reply src/UI/Components/Diagram/actions.ts]: hmm, can you comment that somewhere, I don't think I'll remember that one in a few months

File: changelogs/unreleased/5335-composer-bundle.yml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+change-type: patch
+description: "Fix Test Flake, add r/rw block to modal form"
+issue-nr: 5335
+destination-branches:
+  - master
+  - iso6
```
Comment (LukasStordeur) [inline changelogs/unreleased/5335-composer-bundle.yml:6]: iso7

File: src/UI/Components/Diagram/actions.ts [hunk 7]
```diff
@@ -259,7 +301,7 @@ export function appendInstance(
  * @param {EmbeddedEntity} embeddedEntity that we want to display
  * @param {InstanceAttributeModel} entityAttributes - attributes of given entity
  * @param {string | null} embeddedTo - id of the entity/shape in which this shape is embedded
- * @param {string} holderType - name of the entity type in which is embedded
+ * @param {string} holderName - name of the entity type in which is embedded
```
Comment (LukasStordeur) [inline src/UI/Components/Diagram/actions.ts:304]: name or type are not the same for me, also, maybe parentEntityType or parentEntityName might be clearer? 
Either the description should be 
```suggestion
 * @param {string} holderName - name of the entity to which it is embedded/connected
```
```suggestion
 * @param {string} holderType - Type of the entity to which it is embedded/connected
```

**Pull request closed without merging**